src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d.MM.yyyy");
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        String [] getSalary = getSalaryForEach(data,dateFrom,dateTo);
+        result.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo).append("\n");
+        int counter = 0;
+        for (String name : names) {
+            result.append(name).append(" - ");
+            result.append(getSalary[counter]).append("\n");
+            counter++;
+        }
+        return result.toString().trim();
+    }
+
+    public String[] getSalaryForEach(String[] data, String dateFrom, String dateTo) {
+        int salaryAndrew = 0;
+        int salaryJohn = 0;
+        int salaryKate = 0;
```

Comment:
let's not hardcode these values, cause if names change we will need to modify the code. Instead use names array. You can iterate this arr with for each loop. and for each name run loop with data and compara name with data.[index of name] if yes comapre date and if everything is ok calculate salary for this day and add it to total salary. 
Try this approach, if you have problems come to QnA and we will help you)

Reply:
done